frappe/desk/doctype/todo/todo.py

```diff
@@ -41,6 +43,39 @@ def on_update(self):
 
 		self.update_in_reference()
 
+	def make_recurred_todo(self):
+		startdate = getdate()
+		date = getdate(self.date)
+
+		if date == startdate:
+			frappe.throw(_("Due date should not be today"))
+
+		if self.frequency == 'Weekly':
+			startdate = add_to_date(startdate, days=get_weekdate(startdate, self.day_of_week))
+
+		days_interval = get_interval_days(self.frequency) if self.frequency != "Monthly" else 0
+		months_interval = 1 if self.frequency == "Monthly" else 0
+
+		iterdate = startdate
+
+		while iterdate < date:
+			iterdate = add_to_date(iterdate, days=days_interval, months=months_interval)
+
+			if self.frequency == 'Weekdays' and is_weekend(iterdate):
+				continue
+
+			frappe.get_doc({
```

Comment:
You may also want to copy over some other fields (like "Priority", "Assigned By", "Allocated To")